lib/icingadb/icingadb-utility.cpp

```diff
@@ -245,6 +245,21 @@ String IcingaDB::GetLowerCaseTypeNameDB(const ConfigObject::Ptr& obj)
 }
 
 long long IcingaDB::TimestampToMilliseconds(double timestamp) {
+	// In addition to the limits of the Icinga DB MySQL (0 - 2^64) and PostgreSQL (0 - 2^63) schemata,
+	// years not fitting in YYYY may cause problems, see e.g. https://github.com/golang/go/issues/4556.
+	// RFC 3339: "All dates and times are assumed to be (...) somewhere between 0000AD and 9999AD."
+	//
+	// The below upper limit includes a safety buffer to make sure the timestamp is within 9999AD in all time zones:
+	// $ date -ud @253402214400
+	// Fri Dec 31 00:00:00 UTC 9999
+	// $ TZ=Asia/Vladivostok date -d @253402214400
+	// Fri Dec 31 10:00:00 +10 9999
```

Comment:
By the way, according to [this list from Wikipedia](https://en.wikipedia.org/wiki/List_of_tz_database_time_zones), currently `Pacific/Kiritimati` has the largest UTC offset with UTC+14. (Just rounding that up to a full 24 hours sounds sane to me.[^1])

[^1]: Still has some room for countries jumping over the international date line. As long as we don't get to African or European countries doing that, things are good.

Reply:
Yep. Exactly the "safety buffer to make sure the timestamp is within 9999AD" I wrote about.